components/board/client.go

```diff
@@ -253,6 +265,125 @@ func (dic *digitalInterruptClient) RemoveCallback(c chan Tick) {
 	panic(errUnimplemented)
 }
 
+func (c *client) StreamTicks(ctx context.Context, interrupts []string, ch chan Tick, extra map[string]interface{}) error {
+	c.streamMu.Lock()
+	defer c.streamMu.Unlock()
+	ext, err := protoutils.StructToStructPb(extra)
+	if err != nil {
+		return err
+	}
+	c.extra = ext
```

Comment:
:grimacing: Not sure about storing these in as large a struct as `client`. Suppose I want to subscribe to two different interrupts in two different places (e.g., I have an ultrasonic sensor and an accelerometer that can detect freefall). Wouldn't the second stream destroy the first one? (I barely remember how `client` is supposed to be used; perhaps  this is a nonissue and I'm just not thinking about it right.)

If it _is_ possible you should be able to have multiple interrupt streams simultaneously, consider making an `InterruptStream` class that contains most of the implementation (it stores `extra`, has the mutex(es), the background worker, etc.), and then the `client` struct stores a list of `InterruptStreams` or something (plus a way to remove them from the list when they finish).